File: ArtIsPain/Server/ViewModels/Band/BandViewModel.cs
﻿using ArtIsPain.Server.ViewModels.Album;
using System;
using System.Collections.Generic;

namespace ArtIsPain.Server.ViewModels.Band
{
    public class BandViewModel : IViewModel
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string FormationDate { get; set; }

        public string DisbandDate { get; set; }

        public ICollection<AlbumPreviewModel> Albums { get; set; }
    }
}

File: ArtIsPain/Client/src/app/models/band/BandViewModel.ts
import { AlbumPreviewModel } from '../album/AlbumPreviewModel';
import { IViewModel } from '../../_interfaces/i-view-model';

export class BandViewModel implements IViewModel {
    Id: string;
    Name: string;
    Description: string;
    FormationDate: Date;
    DisbandDate: Date;
    Albums: AlbumPreviewModel[];
}
